libs/prebuilt/langgraph/prebuilt/interrupt.py

```diff
@@ -92,3 +94,107 @@ class HumanResponse(TypedDict):
+            action_request=ActionRequest(
+                action=tool_name,
+                args=tool_call["args"],
+            ),
+            config=interrupt_config,
+            # TODO: add description, we don't have tool list here unfortunately
+            description="Please review tool call before execution",
+        )
+        response = interrupt([request])
+
+        # resume provided by agent inbox as a list
+        if isinstance(response, list):
+            response = response[0]
+
+        try:
+            response_type = response.get("type")
+        except AttributeError:
+            response_type = "invalid"
+
+        if response_type == "accept" and interrupt_config["allow_accept"]:
+            return tool_call
+        elif response_type == "edit" and interrupt_config["allow_edit"]:
+            return ToolCall(
+                args=cast(ActionRequest, response)["args"]["args"],
+                name=tool_name,
+                id=call_id,
+                type="tool_call",
+            )
+        elif response_type == "response" and interrupt_config["allow_respond"]:
+            return ToolMessage(
+                content=cast(str, response["args"]),
+                name=tool_name,
+                tool_call_id=call_id,
+                status="error",
+            )
+        elif response_type == "ignore" and interrupt_config["allow_ignore"]:
+            return ToolMessage(
+                content=f"User ignored the tool call: {tool_name}:{call_id}",
+                name=tool_name,
+                tool_call_id=call_id,
+                status="success",
+            )
+
+        allowed_types = [
+            type_name
+            for type_name, is_allowed in {
+                "accept": interrupt_config["allow_accept"],
+                "edit": interrupt_config["allow_edit"],
+                "response": interrupt_config["allow_respond"],
+                "ignore": interrupt_config["allow_ignore"],
+            }.items()
+            if is_allowed
+        ]
+        raise ValueError(
+            f"Unexpected human response: {response}."
+            f"Expected one with `'type'` in {allowed_types}."
+        )
+
+    def _func(self, input: dict[str, Any]) -> Command | None:
+        ai_msg = input["messages"][-1]
+        tool_calls: list[ToolCall] = ai_msg.tool_calls or []
+        tool_messages: list[ToolMessage] = []
+        pending_tool_calls: bool = False
+        for idx, tool_call in enumerate(tool_calls):
+            if interrupt_config := self.interrupt_policy.get(tool_call["name"]):
+                interrupt_result = self._interrupt(
+                    tool_call=tool_call, interrupt_config=interrupt_config
+                )
+
+                if isinstance(interrupt_result, ToolMessage):
+                    tool_messages.append(interrupt_result)
+                else:
+                    pending_tool_calls = True
+                    tool_calls[idx] = interrupt_result
```

Comment:
so this is modifying in place in ai_msg? could we make a copy to be on the safe side?